MediaContent/Model/ContentProcessor.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaContent\Model;
+
+use Magento\Framework\Exception\IntegrationException;
+use Magento\MediaContentApi\Api\ExtractAssetFromContentInterface;
+use Magento\MediaGalleryApi\Api\Data\AssetInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Process relation managing between media asset and content: assign or unassign relation if exists
+ */
+class ContentProcessor
+{
+    /**
+     * @var ExtractAssetFromContentInterface
+     */
+    private $extractAssetFromContent;
+
+    /**
+     * @var AssignAsset
+     */
+    private $assignAsset;
+
+    /**
+     * @var GetAssetsUsedInContent
+     */
+    private $getAssetsUsedInContent;
+
+    /**
+     * @var UnassignAsset
+     */
+    private $unassignAsset;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * ContentProcessor constructor.
+     *
+     * @param ExtractAssetFromContentInterface $extractAssetFromContent
+     * @param AssignAsset $assignAsset
+     * @param GetAssetsUsedInContent $getAssetsUsedInContent
+     * @param UnassignAsset $unassignAsset
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        ExtractAssetFromContentInterface $extractAssetFromContent,
+        AssignAsset $assignAsset,
+        GetAssetsUsedInContent $getAssetsUsedInContent,
+        UnassignAsset $unassignAsset,
+        LoggerInterface $logger
+    ) {
+        $this->extractAssetFromContent = $extractAssetFromContent;
+        $this->assignAsset = $assignAsset;
+        $this->getAssetsUsedInContent = $getAssetsUsedInContent;
+        $this->unassignAsset = $unassignAsset;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param string $contentEntityId
```

Comment:
please add method description to all introduced methods

Reply:
I think that adding description, for example, to the observer with the one execute method will duplicate class description which already has a meaningful description of the class action. For example, 
```text
/**
 * Observe the catalog_category_save_after event and run processing relation between category content and media asset
 */
```
is a description of action provided by the class. If we will have a method description they both will be identical. Maybe we can remove doc block from the class and use only method description in case when there is a straight forward logic with one method?

As an exception, I can say that ContentProcessor requires adding a description to the method.